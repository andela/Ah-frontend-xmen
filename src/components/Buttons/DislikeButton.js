import React from 'react';

const DislikeButton = (props) => {
  const classNames = props.disliked ? 'btn btn-sm btn-default' : 'btn btn-sm btn-outline-default';
  return (
    <button type="button" id="like-btn" onClick={props.handleDislike} className={classNames}>
      <span className="pr-1"><i className="fa fa-sm fa-thumbs-down" /></span>
      {props.dislikesCount}
    </button>
  );
};

export default DislikeButton;
