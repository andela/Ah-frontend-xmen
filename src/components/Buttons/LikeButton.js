import React from 'react';

const LikeButton = (props) => {
  const classNames = props.liked ? 'btn btn-sm btn-default' : 'btn btn-sm btn-outline-default';
  return (
    <button type="button" id="like-btn" onClick={props.handleLike} className={classNames}>
      <span className="pr-1"><i className="fa fa-sm fa-thumbs-up" /></span>
      {props.likesCount}
    </button>
  );
};

export default LikeButton;
