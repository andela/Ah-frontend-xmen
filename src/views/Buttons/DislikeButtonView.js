import React from 'react';
import { connect } from 'react-redux';
import { toast } from 'react-toastify';
import DislikeButton from '../../components/Buttons/DislikeButton';
import likeDislikeArticleAction from '../../actions/articleActions/LikeDislikeArticleAction';
import getLikeDislikeArticleStatus from '../../actions/articleActions/getLikeDislikeArticleStatus';

export class DislikeButtonView extends React.Component {
  constructor(props) {
    super(props);
    this.handleDislike = this.handleDislike.bind(this);
  }

  componentWillMount() {
    this.props.getDislikeStatus(this.props.slug);
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.dislikeState.errorMessage.dislike) {
      toast.error(nextProps.dislikeState.errorMessage.dislike);
    }
  }

    handleDislike = () => {
      const { slug, isLikeBtn } = this.props;
      const { disliked } = this.props.dislikeState;
      this.props.dislikeAction(slug, !disliked, isLikeBtn);
    }

    render() {
      const { disliked, dislikesCount } = this.props.dislikeState;

      return (
        <DislikeButton
          disliked={disliked}
          handleDislike={this.handleDislike}
          dislikesCount={dislikesCount}
        />
      );
    }
}

const mapStateToProps = state => ({
  dislikeState: state.likeArticleReducer,
});

const actionCreators = {
  dislikeAction: likeDislikeArticleAction,
  getDislikeStatus: getLikeDislikeArticleStatus,
};

export default connect(
  mapStateToProps,
  actionCreators,
)(DislikeButtonView);
