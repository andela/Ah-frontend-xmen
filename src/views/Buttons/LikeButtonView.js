import React from 'react';
import { connect } from 'react-redux';
import LikeButton from '../../components/Buttons/LikeButton';
import likeDislikeArticleAction from '../../actions/articleActions/LikeDislikeArticleAction';
import getLikeDislikeArticleStatus from '../../actions/articleActions/getLikeDislikeArticleStatus';

export class LikeButtonView extends React.Component {
  constructor(props) {
    super(props);
    this.handleLike = this.handleLike.bind(this);
  }

  componentWillMount() {
    this.props.getLikeStatus(this.props.slug);
  }


    handleLike = () => {
      const { slug, isLikeBtn } = this.props;
      const { liked } = this.props.likeState;
      this.props.likeAction(slug, !liked, isLikeBtn);
    }

    render() {
      const { liked, likesCount } = this.props.likeState;
      return (
        <LikeButton
          liked={liked}
          handleLike={this.handleLike}
          likesCount={likesCount}
        />
      );
    }
}

const mapStateToProps = state => ({
  likeState: state.likeArticleReducer,
});
const actionCreators = {
  likeAction: likeDislikeArticleAction,
  getLikeStatus: getLikeDislikeArticleStatus,
};
export default connect(
  mapStateToProps, actionCreators,
)(LikeButtonView);
