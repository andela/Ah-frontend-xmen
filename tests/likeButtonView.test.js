import React from 'react';
import { shallow } from 'enzyme';
import { LikeButtonView } from '../src/views/Buttons/LikeButtonView';


describe('LikeButtonView', () => {
  it('should render without crashing', () => {
    const props = {
      getLikeStatus: jest.fn(),
      likeState: {
        liked: false,
        likesCount: 0,
      },
    };
    const wrapper = shallow(<LikeButtonView {...props} />);
    expect(wrapper).toMatchSnapshot();
  });


  it('should call dispatch actionon handlelike', () => {
    const props = {
      slug: 'slug-213',
      isLikeBtn: true,
      likeAction: jest.fn(),
      getLikeStatus: jest.fn(),
      likeState: {
        liked: false,
        likesCount: 0,
      },
    };

    const wrapper = shallow(<LikeButtonView {...props} />);
    wrapper.instance().handleLike();
    expect(wrapper.instance().props.likeAction).toHaveBeenCalled();
  });
});
