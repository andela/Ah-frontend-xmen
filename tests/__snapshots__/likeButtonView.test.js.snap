// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LikeButtonView should render without crashing 1`] = `ShallowWrapper {}`;
